### PR TITLE
[artifactory][artifactory-ha] Allow overriding the urlStrictPolicy setting for webhooks

### PR DESCRIPTION
**Pull request by KierranM**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [x] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:

This allows setting the `event.webhooks.urlStrictPolicy` setting, this is needed in environments where the webhook target is private. Without this you are unable to save a new webhook pointing at an internal service and the logs throw this error:
```
error validating webhook url: illegal private network: <internal ip here> [subscriptions]
```

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

I wasn't sure how to (or whether to) do the version bump since the changes to align the chart version with the app version.

### CODE REVIEW

**Comment by chukka** [comment]
@KierranM  Thanks for this contribution , we would like to review this internally and take it forward from there, This change will be part of 7.19.x release. Note: No further is action is needed from you .

File: stable/artifactory-ha/values.yaml [hunk 1]
```diff
@@ -253,6 +253,11 @@ artifactory:
   openMetrics:
     enabled: false
 
+  webhooks:
```
Comment (chukka) [inline stable/artifactory-ha/values.yaml:256]: @KierranM  please change this to events.webhooks.urlStrictPolicy : true

File: stable/artifactory-ha/values.yaml [hunk 2]
```diff
@@ -610,6 +615,9 @@ artifactory:
     replicator:
       enabled: true
     {{- end }}
+    event:
+      webhooks:
+        urlStrictPolicy: {{ .Values.artifactory.webhooks.urlStrictPolicy }}
```
Comment (chukka) [inline stable/artifactory-ha/values.yaml:620]: please change this to .Values.artifactory.events.webhooks.urlStrictPolicy

File: stable/artifactory/values.yaml [hunk 3]
```diff
@@ -539,6 +544,9 @@ artifactory:
     replicator:
       enabled: true
     {{- end }}
+    event:
+      webhooks:
+        urlStrictPolicy: {{ .Values.artifactory.webhooks.urlStrictPolicy }}
```
Comment (chukka) [inline stable/artifactory/values.yaml:549]: please change this to .Values.artifactory.events.webhooks.urlStrictPolicy

File: stable/artifactory/values.yaml [hunk 4]
```diff
@@ -209,6 +209,11 @@ artifactory:
   openMetrics:
     enabled: false
 
+  webhooks:
```
Comment (chukka) [inline stable/artifactory/values.yaml:212]: please change this to .events.webhooks.urlStrictPolicy: true

**Review by chukka — COMMENTED** [review]
@KierranM Thanks for this contribution , Can you address these review comments, Post that we would like to review this internally and take it forward from there, This change will be part of 7.19.x release. Note: No further action will be  needed from you post these review comments

**Review by chukka — CHANGES_REQUESTED** [review]
@KierranM just verified  with https://www.jfrog.com/confluence/display/JFROG/Artifactory+System+YAML , event should be root element (not under artifactory), Can you make corresponding changes

**Comment by chukka** [comment]
@KierranM Also , thinking if we can set this to default false ? customers can enable by setting this to true as needed.

Also, looking at systemyaml configuration, by default it was true, was it not working with previous charts with adding any configuration ?

**Comment by chukka** [comment]
@KierranM Can you please verify if it works without this PR (as default configuration is to allow `urlStrictPolicy` to true)

**Comment by KierranM** [comment]
> @KierranM just verified with https://www.jfrog.com/confluence/display/JFROG/Artifactory+System+YAML , event should be root element (not under artifactory), Can you make corresponding changes

I'm currently away from a computer for the next few days, but I'll update this as soon as I'm back

**Comment by KierranM** [comment]
> @KierranM Can you please verify if it works without this PR (as default configuration is to allow `urlStrictPolicy` to true)

The issue we're facing is _because_ it defaults to `true`. We need to be able to disable this setting otherwise we can't create webhooks pointing at internal APIs

**Comment by chukka** [comment]
> > @KierranM just verified with https://www.jfrog.com/confluence/display/JFROG/Artifactory+System+YAML , event should be root element (not under artifactory), Can you make corresponding changes
> 
> I'm currently away from a computer for the next few days, but I'll update this as soon as I'm back

Sure

**Comment by chukka** [comment]
@KierranM If you would like to modify default systemyaml (defined in values.yaml), you can make use of [systemYamlOverride](https://github.com/jfrog/charts/blob/master/stable/artifactory/values.yaml#L66) feature .I'm closing this

**Comment by KierranM** [comment]
@chukka, it's true we _can_ use the override, but you can only override the _entire_ system yaml. Doing it this way means we have to check the source chart every time we do an upgrade (either chart version or artifactory version) and make sure we copy over any changes that have been made here. I'm concerned about the amount of potential for errors this manual process introduces.

Is there a particular reason you aren't willing to allow this to be configurable as a chart parameter?

**Comment by chukka** [comment]
@KierranM I think `systemYamlOverride` approach is for advanced users. However , Its only one time configuration change and doesn't impact with chart changes going forward, you know what values can be in system.yaml . I prefer this approach  as you have complete control over systemyaml changes.

Our current systemyaml is already very lengthy and wouldn't want to overload with additional changes.

**Pull request closed without merging**